src/data/reports/mal-data-subscription/MalDataSubscriptionDefaultRepository.ts

```diff
@@ -79,98 +28,361 @@ export class MalDataSubscriptionDefaultRepository implements MalDataSubscription
         this.globalStorageClient = new DataStoreStorageClient("global", instance);
     }
 
-    async get(options: MalDataSubscriptionOptions): Promise<PaginatedObjects<MalDataSubscriptionItem>> {
-        const { config, elementTypes, dataElementIds, sections } = options; // ?
-        const { sorting, paging } = options; // ?
-
-        const sqlViews = new Dhis2SqlViews(this.api);
-        const allDataSetIds = _.values(config.dataSets).map(ds => ds.id); // ?
-
-        const { pager, rows } = await sqlViews
-            .query<Variables, SqlField>(
-                getSqlViewId(config, SQL_VIEW_MAL_DATAELEMENTS_NAME),
-                {
-                    dataSets: sqlViewJoinIds(allDataSetIds),
-                    elementType: sqlViewJoinIds(elementTypes),
-                    sectionId: sqlViewJoinIds(sections),
-                    dataElementId: sqlViewJoinIds(dataElementIds),
-                    orderByColumn: fieldMapping[sorting.field],
-                    orderByDirection: sorting.direction,
-                },
-                paging
+    async get(
+        options: MalDataSubscriptionOptions
+    ): Promise<MalSubscriptionPaginatedObjects<DataElementsSubscriptionItem>> {
+        const {
+            dataElementGroups: dataElementGroupIds,
+            subscriptionStatus,
+            sections: sectionIds,
+            sorting,
+            paging,
+        } = options;
+        if (!sorting)
```

Comment:
To improve performance and avoid unnecessary api calls here you can check also elementType !== "dataElements"